### ISSUE TITLE
pass in object to be serialized to json and sent?

### ISSUE THREAD

**Issue by jnystrom** [Mimeo/SendWithUs.Client#3]
I see you are taking a dictionary of data, any thought to taking in a type T and serializing that to json and sending that over to SendWithUs?

**Comment by lencharest** [comment]
On the wire, the value of the `email_data` property in the request must be a JSON object, and an arbitrary type T wouldn't necessarily serialize as that; e.g., for T = `List<string>`, we'd get a JSON array.

Are you looking for complex data values? I could change the type of `SendRequest.Data` to `IDictionary<string, object>` to support that.

**Comment by lencharest** [comment]
After noodling on this for a while, I've decided to change the type of `ISendRequest.Data` to plain old `object`. This will get you what you want, and the necessary code changes are minimal. However, the onus of ensuring a proper JSON serialization will be on whoever sets the `Data` property (i.e., you). I'm just going to pass the value of `Data` directly to the JSON serializer.

For convenience, I'm also going to define `SendRequest<T>` with a `Data` property of type T that shadows the `Data` property on the non-generic `SendRequest` class.

**Comment by jnystrom** [comment]
Yes, I was about to use your code and give this ability, but you are correct the responsibility of the data and the correct structure would be on me.  The responsibility of it was same before, because I still had to add the keys correctly to the dictionary.
